src/solutions/ode_solutions.jl

```diff
@@ -115,6 +115,8 @@ struct ODESolution{T, N, uType, uType2, DType, tType, rateType, P, A, IType, S,
     stats::S
     alg_choice::AC
     retcode::ReturnCode.T
+    resid::R
+    original::O
```

Comment:
What about creating a ```BVPSolution``` to specifically handle these details only in the BVP context?

Reply:
I am open to that. We can do that by subtyping AbstractODESolution.

Reply:
IMO it just seems a bit wasteful code wise. It's just two more fields, and `original` makes sense in the IVP case as well (we should use it with ODEInterface)